myJoin.pl

```diff
@@ -98,33 +103,28 @@ sub line_hash1{
         @headers = split("\t", $header_text);
         %header_lib = map {$headers[$_] => $_} 0..$#headers;
     }
-#    print "reading $input\n";
+#     print "headers: @headers\n";
+# 	  print Dumper %header_lib;
     while (<FILE>){
+    	chomp;
```

Comment:
You should **almost always** `chomp` each line of a file when reading it, and then add the newlines back in later if needed. This greatly cleans up the data you are working with, reducing surprises down the line.